src/test/java/com/heatheryou/httpserver/RouterTest.java

```diff
@@ -21,38 +23,11 @@ public void allowedMethodsReturnsListOfAvailableMethodsGivenUri() {
     }
 
     @Test
-    public void isValidRequestReturnsTrueIfRequestedMethodAndUriAreSupported() {
+    public void handleRequestReturnsGetHandlerIfValidUriWithGETMethodRequested() {
         Router router = new Router();
-        boolean actual = router.isValidRequest("/simple_get", "HEAD");
-        assertTrue(actual);
-    }
-
-    @Test
-    public void isValidRequestReturnsFalseIfRequestedUriIsValidButMethodNotSupported() {
-        Router router = new Router();
-        boolean actual = router.isValidRequest("/simple_get", "POST");
-        assertFalse(actual);
-    }
-
-    @Test
-    public void isValidRequestReturnsFalseIfRequestedUriAndMethodNotSupported() {
-        Router router = new Router();
-        boolean actual = router.isValidRequest("/invalid_uri", "POST");
-        assertFalse(actual);
-    }
-
-    @Test
-    public void isValidRouteReturnsTrueIfValidUriIsRequested() {
-        Router router = new Router();
-        boolean actual = router.isValidRoute("/simple_get");
-        assertTrue(actual);
-    }
-
-    @Test
-    public void isValidRouteReturnsFalseIfInvalidUriIsRequested() {
-        Router router = new Router();
-        boolean actual = router.isValidRoute("/invalid_uri");
-        assertFalse(actual);
+        Request request = new Request("/simple_get", "GET");
+        RequestHandler actual = router.handleRequest(request);
+        assertThat(actual, instanceOf(GetHandler.class));
```

Comment:
This an example of something that makes tests a bit brittle. The verification here is of form and not behavior. At every point possible, tests should target verifying behavior rather than type/shape/etc. This is primarily because, if your types ever change, or if the handlers become more polymorphic (or get abstracted via interfaces), this test would need to change. Ideally, if tests verify behavior, adding functionality should not result in failures of existing behavior. (This isn't always possible. If your framework is nothing but concrete types and lots of tiny specifics about using it, sometimes you really do have to write tests like this. That being said, you control all the code here. :smiley: )

Reply:
This makes sense! I guess with these tests, I just wanted to make sure the correct handler was being called. Now that I have tested that, these tests will be gone in the next PR! 🙂